formulae/viewer/routes.py

```diff
@@ -0,0 +1,205 @@
+
+            #avoid error when the number of view is under the mainimum
+            elif view<0:
+
+
+                view=0
+                flash( _('There are not {} images for this formula. Showing the first page image instead.'.format(view)))
+                if "/images/formulae-1" in str(images[view]):
+                    link_picture=current_app.IIIFserver +str(images[view])
+                else:
+                    link_picture=str(images[view])
+                return current_app.config['nemo_app'].render(template="viewer::newtabviewer.html", picture=link_picture, objectId=objectId,
+                                                         current_view=view, total_views=len(images), codex=codex, folios=folios, url=dict())
+
+            #create the link if it's an image in local server or take the complet link
+            if "/images/formulae-1" in str(images[view]):
+                link_picture=current_app.IIIFserver +str(images[view])
+            else:
+                link_picture=str(images[view])
+
+            return current_app.config['nemo_app'].render(template="viewer::newtabviewer.html", picture=link_picture, objectId=objectId,
+                                                         current_view=view, total_views=len(images), codex=codex, folios=folios, url=dict())
+
+    else:
+        flash( _('This corpus is on copyright, please choose another text'))
+        return current_app.config['nemo_app'].render(template='main::index.html', url=dict())
+
+
+@bp.route("embedded/<objectId>/<view>", methods=["GET"])
+def addviewer(objectId, view):
+
+    if current_app.config['nemo_app'].check_project_team() is True or objectId in current_app.config['nemo_app'].open_texts:
+
+        #information need for loris or
+        collection = current_app.config['nemo_app'].get_collection(objectId)
+        if isinstance(collection, CtsWorkMetadata):
+            editions = [t for t in collection.children.values() if isinstance(t, CtsEditionMetadata)]
+            if len(editions) == 0:
+                raise UnknownCollection('{}'.format(collection.get_label()) + _l(' hat keine Edition.'))
+            objectId = str(editions[0].id)
+        formulae = current_app.picture_file[objectId]
+        passage_data = get_passage(objectId, '1')
+
+
+        #this viewer work when the library or archiv give an IIIF API for the external usage of theirs books
+        if "manifest" in formulae:
+            if "codex" in formulae:
+                codex=formulae["codex"]
+                manifest = url_for('viewer.static', filename=formulae["manifest"])
+                return current_app.config['nemo_app'].render(template='viewer::multiviewermirador.html', manifest=manifest
+                                                             ,objectId=objectId, codex=codex, passage_data=passage_data, url=dict())
+            else:
+                manifest = url_for('viewer.static', filename=formulae["manifest"])
```

Comment:
Since lines 125 and 121 are identical, I would combine them into a single command after line 118 (`if "manifest" in formulae:`).